Mathlib/NumberTheory/NumberField/Discriminant/Different.lean

```diff
@@ -61,11 +63,45 @@ lemma NumberField.absNorm_differentIdeal : (differentIdeal ℤ 𝒪).absNorm = (
     simp [b', Module.Basis.toMatrix_apply, mul_comm (RingOfIntegers.basis K i),
       b, integralBasis_apply, ← map_mul, Algebra.trace_localization ℤ ℤ⁰]
 
-lemma NumberField.discr_mem_differentIdeal : ↑(discr K) ∈ differentIdeal ℤ 𝒪 := by
+lemma NumberField.discr_mem_differentIdeal [Module.Free ℤ 𝒪] [IsDedekindDomain 𝒪] :
+    ↑(discr K) ∈ differentIdeal ℤ 𝒪 := by
   have := (differentIdeal ℤ 𝒪).absNorm_mem
   cases (discr K).natAbs_eq with
   | inl h =>
     rwa [absNorm_differentIdeal (K := K), ← Int.cast_natCast, ← h] at this
   | inr h =>
     rwa [absNorm_differentIdeal (K := K), ← Int.cast_natCast, Int.eq_neg_comm.mp h,
       Int.cast_neg, neg_mem_iff] at this
+
+lemma NumberField.not_dvd_discr_iff_forall_liesOver {p : ℤ} (hp : Prime p) :
+    ¬ p ∣ discr K ↔ ∀ (P : Ideal 𝒪) (_ : P.IsMaximal), P.LiesOver (.span {p}) →
+      Algebra.IsUnramifiedAt ℤ P := by
```

Comment:
Maybe we should introduce something like `Algebra.IsUnramifiedOver` that says what this long line of implication is saying